File: pset2/ceasar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>


int main(int argC, string argV[]){
    string plaintext;
    int key;
    int plaintext_len;

    if(argC >= 2){
        // Ask plainext from user
        plaintext = get_string("plaintext:  ");
        // Get plaintext length
        plaintext_len = strlen(plaintext);
        // Convert key to integer
        key = atoi(argV[1]);
        printf("ciphertext: ");
        // Encrypt plaintext
        for(int i=0; i < plaintext_len; i++){
            char encrypt_char;

            if(islower(plaintext[i]))
                encrypt_char = ((plaintext[i] + key - 97) % 26) + 97;

            else if(isupper(plaintext[i]))
                encrypt_char = ((plaintext[i] + key - 65) % 26) + 65;

            else
                encrypt_char = plaintext[i];

            printf("%c", encrypt_char);
        }
        printf("\n");
    }else{
        return 1;
    }

    return 0;
}

File: pset6/caesar/caesar.py

# A program that encrypts messages using Caesar’s cipher, per the below.
# See CS50 doc for more information
#
# Author: Jeanniton Monero
# email: jeanniton.mnr@gmail.com
# website: jeanniton.me

import sys


def main():

    paintext = ""
    key = int(sys.argv[1])
    plaintext_len = None


    if key >= 0:
        # Get plaintext from user
        plaintext = input("Plaintext: ")
        # Save plaintext length
        plaintext_len = len(plaintext)

        # Encrypt plaintext
        print("ciphertext: ", end="")
        for i in range(plaintext_len):
            # Get the ASCII number of a character
            char = plaintext[i]
            char_index = ord(char)
            encrypt_char = None
            if char.islower():
                # ord('a') = 97
                encrypt_char = ((char_index + key - 97) % 26) + 97
            elif char.isupper():
                # ord('A') = 65
                encrypt_char = ((char_index + key -65) %26) + 65
            else:
                encrypt_char = char_index
            # Print the character given by an ASCII number
            print(chr(encrypt_char), end="")
        # Print new line
        print("")

        return 0
    else:
        return 1



if __name__ == "__main__":
    main()